src/gpaletteeditor.c

```diff
@@ -619,7 +619,7 @@ void setPaletteName(char* paletteName, int index)
 	{
 		free(paletteNames[index]);
 	}
-	paletteNames[index] = malloc(strlen(paletteName));
+	paletteNames[index] = malloc(strlen(paletteName) + 1);
```

Comment:
Even simpler : replace malloc and strpcy with "strdup(paletteName)".

Reply:
Makes sense! It's already used in other parts of the codebase so there are no compatibility concerns.